qiskit/algorithms/time_evolution/evolution_result.py

```diff
@@ -0,0 +1,39 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Class for holding evolution result and relevant metadata."""
+
+from qiskit.algorithms import AlgorithmResult
+from qiskit.opflow import OperatorBase
+
+
+class EvolutionResult(AlgorithmResult):
+    """Class for holding evolution result and relevant metadata."""
+
+    def __init__(self, evolved_state: OperatorBase = None, evolved_observable: OperatorBase = None):
```

Comment:
This should have the `Optional` qualifier if they can be `None` 🙂 
```suggestion
    def __init__(self, evolved_state: Optional[OperatorBase] = None, evolved_observable: Optional[OperatorBase] = None):
```